hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -679,6 +679,11 @@ public String getResponseHeader(final String httpHeader) {
       }
       return EMPTY_STRING;
     }
+
+    @Override
+    public Map<String, List<String>> getResponseHeaders() {
+      return null;
```

Comment:
Why are we returning null here ?

Reply:
This is needed for the flow where getXAttr() is called on an implicit path. This hard-set result will be returned for an implicit path only and that path cannot have any attributes assigned to it hence null.

Reply:
While resolving merge conflicts I realised pranav has already defined this for apache client work and his implementation will also work so removing this from my patch